analytic_multicurrency/tests/test_analytic_lines.py

```diff
@@ -0,0 +1,489 @@
+
+        self.aajournal = self.ref('account.analytic_journal_sale')
+
+    def test_analytic_lines(self):
+        cr, uid = self.cr, self.uid
+
+        aal_id = self.account_analytic_line_obj.create(cr, uid, {
+            'account_id': self.agrolait,
+            'name': 'AGROLAIT',
+            'journal_id': self.aajournal,
+            'date': time.strftime('%Y-%m-%d'),
+            'amount': 100,
+            'general_account_id': self.account_rcv_id,
+        })
+        aal_brw = self.account_analytic_line_obj.browse(cr, uid, aal_id)
+        self.assertEqual('EUR', aal_brw.aa_currency_id.name)
+        self.assertEqual(100, aal_brw.aa_amount_currency)
+        self.assertEqual(100, aal_brw.account_id.ca_invoiced)
+        self.assertEqual(0, aal_brw.account_id.total_cost)
+
+    def test_analytic_lines2(self):
+        cr, uid = self.cr, self.uid
+        self.res_currency_rate_model.create(cr, uid, {
+            'name': time.strftime('%Y-%m-%d') + ' 00:00:00',
+            'currency_id': self.currency_usd_id,
+            'rate': 0.50,
+        })
+        self.registry('account.analytic.account').write(
+            cr, uid,
+            [self.agrolait],
+            {'currency_id': self.currency_usd_id}
+        )
+        aal_id = self.account_analytic_line_obj.create(cr, uid, {
+            'account_id': self.agrolait,
+            'name': 'AGROLAIT',
+            'journal_id': self.aajournal,
+            'date': time.strftime('%Y-%m-%d'),
+            'amount': 100,
+            'general_account_id': self.account_rcv_id,
+        })
+        aal_brw = self.account_analytic_line_obj.browse(cr, uid, aal_id)
+        self.assertEqual('USD', aal_brw.aa_currency_id.name)
+        self.assertEqual(50, aal_brw.aa_amount_currency)
+
+    def test_analytic_lines3(self):
+        cr, uid = self.cr, self.uid
+        self.res_currency_rate_model.create(cr, uid, {
+            'name': time.strftime('%Y-%m-%d') + ' 00:00:00',
+            'currency_id': self.currency_usd_id,
+            'rate': 0.50,
+        })
+
+        aal_id = self.account_analytic_line_obj.create(cr, uid, {
+            'account_id': self.agrolait,
+            'name': 'AGROLAIT',
+            'journal_id': self.aajournal,
+            'date': time.strftime('%Y-%m-%d'),
+            'amount': 100,
+            'general_account_id': self.account_rcv_id,
+        })
+        aal_brw = self.account_analytic_line_obj.browse(cr, uid, aal_id)
+        self.assertEqual('EUR', aal_brw.aa_currency_id.name)
+        self.assertEqual(100, aal_brw.aa_amount_currency)
+
+        self.registry('account.analytic.account').write(
+            cr, uid,
+            [self.agrolait],
+            {'currency_id': self.currency_usd_id}
+        )
+        aal_brw = self.account_analytic_line_obj.browse(cr, uid, aal_id)
+        self.assertEqual('USD', aal_brw.aa_currency_id.name)
+        self.assertEqual(50, aal_brw.aa_amount_currency)
+
+    def test_analytic_lines4(self):
```

Comment:
I agree that as it is the function name is useless. IMHO, even better would be use a function name that tells what the test does (if it does not fit, that a docstring is of course OK). i.e. 
```python
def test_that_blah_blah_does_blah_blah()
```